frontend/src/tests/components/eMIB/transformations.test.js

```diff
@@ -46,3 +47,44 @@ describe("Check contactNameFromId", () => {
     expect(contactNameFromId(addressBook, 6)).toEqual("");
   });
 });
+
+describe("recursivelyProcessTree", () => {
+  it("returns a processed array for single level tree", () => {
+    const treeContent = [
+      {
+        id: 0,
+        text: "Claude Huard - Manager",
+        team_information_tree_child: []
+      }
+    ];
+    const array = recursivelyProcessTree(treeContent, "team_information_tree_child", 1, 0);
+    expect(array).toEqual([
+      { id: 0, level: 1, name: "Claude Huard - Manager", groups: [], parent: undefined }
+    ]);
+  });
+
+  it("returns a processed array for double level tree", () => {
+    const treeContent = [
+      {
+        id: 0,
+        text: "Claude Huard - Manager",
+        team_information_tree_child: [
+          {
+            text: "Danny McBride - QA Analyst",
+            id: 0
+          },
+          {
+            text: "Serge Duplessis - QA Analyst",
+            id: 1
+          }
+        ]
+      }
+    ];
+    const array = recursivelyProcessTree(treeContent, "team_information_tree_child", 1, 0);
+    expect(array).toEqual([
+      { id: 0, level: 1, name: "Claude Huard - Manager", groups: [1, 2] },
+      { id: 1, level: 2, name: "Danny McBride - QA Analyst", parent: 0 },
+      { id: 2, level: 2, name: "Serge Duplessis - QA Analyst", parent: 0 }
+    ]);
+  });
+});
```

Comment:
Could you please add a test for a triple level tree, since the tree view of Organizational Structure is three levels?

We shouldn't have more than three levels though. Also, maybe we should add a comment saying that your code is working for one level to three levels tree views, unless it's working for four levels and more.

Reply:
My code works for any number of levels so I don't think it's necessary to test all of the cases.